src/coreclr/src/vm/genericdict.cpp

```diff
@@ -71,11 +71,11 @@ DictionaryLayout* DictionaryLayout::Allocate(WORD              numSlots,
 
 //---------------------------------------------------------------------------------------
 //
-// Count the number of bytes that are required in a dictionary with the specified layout
+// Count the number of bytes that are required for slots in a dictionary with the specified layout
 // 
 //static
 DWORD 
-DictionaryLayout::GetDictionarySizeFromLayout(
+DictionaryLayout::GetDictionarySlotSizeFromLayout(
```

Comment:
I would make it one method with `DWORD* ppSlotSize` argument to make sure that you always get consistent snapshot for slot size and alloc size. 

This method is sometimes called without locks where one thread is reading the layout and other thread is expanding it. It is why I had `Volatile...` in my code snippet. Having one method to return both alloc and slot sizes will avoid bugs with somebody accidentally changes the order in which  `GetDictionarySlotSizeFromLayout` and `GetDictionaryAllocSizeFromLayout` are called.

Reply:
Good point, sorry about not realizing this, fixed in 6th commit.